File: shared/contracts/requests.ts
export type RoomCreateRequest = {
  nickname: string;
};

export type RoomJoinRequest = {
  roomCode: string;
  nickname: string;
};

export type RoomReadyRequest = {
  isReady: boolean;
};

export type GameStartRequest = {
  requestId?: string;
};

export type PlayCardRequest = {
  requestId?: string;
  cardId: string;
  targetPlayerId?: string;
};

export type DrawCardRequest = {
  requestId?: string;
};

export type ReconnectRequest = {
  playerSessionId: string;
};

File: backend/app/schemas/requests.py
from pydantic import BaseModel, ConfigDict


class SocketRequestModel(BaseModel):
    model_config = ConfigDict(extra="forbid")


class RoomCreateRequest(SocketRequestModel):
    nickname: str


class RoomJoinRequest(SocketRequestModel):
    roomCode: str
    nickname: str


class RoomReadyRequest(SocketRequestModel):
    isReady: bool


class GameStartRequest(SocketRequestModel):
    requestId: str | None = None


class PlayCardRequest(SocketRequestModel):
    requestId: str | None = None
    cardId: str
    targetPlayerId: str | None = None


class DrawCardRequest(SocketRequestModel):
    requestId: str | None = None


class ReconnectRequest(SocketRequestModel):
    playerSessionId: str
